### ISSUE TITLE
Spike: Determine which WCS flows need E2E tests

### ISSUE THREAD

**Issue by c-shultz** [Automattic/woocommerce-services#2002]

**Comment by c-shultz** [comment]
@Ferdev, already created some good issues for this, so these should be included in the master thread

**Tests:**
https://github.com/Automattic/woocommerce-services/issues/1915
https://github.com/Automattic/woocommerce-services/issues/1918
https://github.com/Automattic/woocommerce-services/issues/1917

**Infrastructure Improvements:**
https://github.com/Automattic/woocommerce-services/issues/1914
https://github.com/Automattic/woocommerce-services/issues/1898

**Comment by c-shultz** [comment]
For getting started with Looker, there's some good in the comments here: https://github.com/Automattic/woocommerce-shipping-issues/issues/22

**Comment by harriswong** [comment]
I agree to focusing on identifying and testing the critical flows, but not every use-cases. End to end test takes long time to run and it also has a high maintenance cost. I vote for the "test pyramid" approach where we focus more on the bottom of the pyramid than the top.

**Comment by c-shultz** [comment]
As part of this, let's also look at what Proton has done to run core tests in Docker to make debugging tests much easier. We should determine how easy this would be for us to adopt and how much value we think we can get from it.

**Comment by c-shultz** [comment]
+1 to focusing on the critical flows. Let's also update the linked issues as part of this to be really clear about what steps/flow the E2E should follow, so none of our PRs get lost in testing every possible use case.

#1915
#1918
#1917
#1916

**Comment by Ferdev** [comment]
hey @c-shultz, was this comment intended for another issue? https://github.com/Automattic/woocommerce-services/issues/2002#issuecomment-613142168

**Comment by c-shultz** [comment]
Lol. yep, I'm not sure what happened there. I had a few issues open at the same time.

**Comment by harriswong** [comment]
### User flow:
A user trying to accomplish a goal and how this user click through WCS. Some of these user flow are the following. If any of these failed, the user can't buy label and that means WCS is not working properly:
- Purchasing label
- Reprinting label
- Refunding label
	- Purchasing label again after refund
- Track label
- WCS settings
- WCS banner from WC

### Order status
How the user interact with WCS also depends on the order status. New order and existing order should provide different user flow. Some of the following overlaps the user flow listed above. But I wanted to list them out and consolidate after.
#### New order
- Click create shipping label, click buy shipping label
- Shipping label modal
	- Filling the form and click "buy shipping label"
	- Test 1 item 
	- Test multiple items with multiple packages
Manual create order
- Click crete shipping label, but manual enter addresses
- Repeat above test cases

#### Existing order
- Shipping label model
	- Should display Tracking
- Shipment Tracking	
	- View details -> should popup
	- Request refund
	- Reprint
